### ISSUE TITLE
Error Seaching Sql Server 2014 [Incorrect syntax near *]

### ISSUE THREAD

**Issue by bitoliveira** [yajra/laravel-datatables#352]
Hello.

I'm having a problem seaching.

I use
- SQL Server 2014 Express
- Laravel 5.2
- yajra/laravel-datatables-oracle (v6.3.1)

```php
SQLSTATE[42000]: [Microsoft][ODBC Driver 11 for SQL Server][SQL Server]Incorrect syntax near '*'. 
(SQL: select count(*) as aggregate from (select '1' as [row_count] from [visa_types] 
where (LOWER([id]) LIKE %p% or LOWER(*) LIKE %p%) and [visa_types].[deleted_at] is null) 
count_row_table)
```

**Comment by yajra** [comment]
Are using `->select('*')` for your query? This code seems `LOWER(*) LIKE %p%` you do.

Anyways, can you try selecting the fields manually? select * may not be fully compatible on SQL Server.

**Comment by bitoliveira** [comment]
First of all sorry for the dealy on the replay. I have to travel. 
I use VisaType::all();
I will try selecting the fields and post again. Thanx

**Comment by IronSinew** [comment]
I had a similar issue with an issue when I was just selecting *. Because I had accessors set up for display (and use) in datatables I couldn't just select the columns individually (without breaking my accessors). 

I ended up taking this approach in `QueryBuilderEngine.php`

```php
protected function compileGlobalSearch($query, $column, $keyword)
    {
        if($column === "*") {
            // Loop through all of the searchable columns and apply filters
            $columns = $this->request->get('columns');
            foreach($columns as $column) {
                if($column['searchable'] == "true") {
                    $this->compileGlobalSearch($query, $column['data'], $keyword);
                }
            }
        } else {
            $column = $this->castColumn($column);
            $sql    = $column . ' LIKE ?';
            if ($this->isCaseInsensitive()) {
                $sql     = 'LOWER(' . $column . ') LIKE ?';
                $keyword = Str::lower($keyword);
            }
            $query->orWhereRaw($sql, [$keyword]);
        }
    }
```

I didn't have a ton of time to look into more of the object methods in order to properly implement this change, but maybe it will be helpful in the future.

**Comment by yajra** [comment]
Hi Guys,

Can you please paste some snippets on how to replicate the issue? 

Thanks!

**Comment by yajra** [comment]
I just posted some updates that [improves column name detection for filtering and ordering](https://github.com/yajra/laravel-datatables/commit/425f65bf260d56ebe23d5cba3256113e12018e78). Can you please try `dev-develop` branch if it fixes this issue? Thanks in advance!

**Comment by yajra** [comment]
I just released v6.6.0 that should fix this issue. Kindly check and confirm if possible. Thanks!
